src/sentry/workflow_engine/migration_helpers/alert_rule.py

```diff
@@ -1,20 +1,152 @@
+                return type[1]
+        return None
+    else:
+        return Action.Type.EMAIL
+
+
+def migrate_metric_action(
+    alert_rule_trigger_action: AlertRuleTriggerAction,
+) -> tuple[Action, DataConditionGroupAction] | None:
+    try:
+        alert_rule_trigger_data_condition = AlertRuleTriggerDataCondition.objects.get(
+            alert_rule_trigger=alert_rule_trigger_action.alert_rule_trigger
+        )
+    except AlertRuleTriggerDataCondition.DoesNotExist:
+        logger.exception(
+            "AlertRuleTriggerDataCondition does not exist",
+            extra={"alert_rule_trigger_id": alert_rule_trigger_action.alert_rule_trigger.id},
+        )
+        return None
+
+    action_type = get_action_type(alert_rule_trigger_action)
+    if not action_type:
+        logger.warning(
+            "Could not find a matching Action.Type for the trigger action",
+            extra={"alert_rule_trigger_action_id": alert_rule_trigger_action.id},
+        )
+        return None
+
+    data = {
+        "type": alert_rule_trigger_action.type,
+        "sentry_app_id": alert_rule_trigger_action.sentry_app_id,
+        "sentry_app_config": alert_rule_trigger_action.sentry_app_config,
+    }
+    action = Action.objects.create(
+        required=False,
+        type=action_type,
+        data=data,
+        integration_id=alert_rule_trigger_action.integration_id,
+        target_display=alert_rule_trigger_action.target_display,
+        target_identifier=alert_rule_trigger_action.target_identifier,
+        target_type=alert_rule_trigger_action.target_type,
+        legacy_notification_type=Action.LegacyNotificationType.METRIC_ALERT,
+    )
+    data_condition_group_action = DataConditionGroupAction.objects.create(
+        condition_group_id=alert_rule_trigger_data_condition.data_condition.condition_group.id,
+        action_id=action.id,
+    )
+    return action, data_condition_group_action
+
+
+def migrate_metric_data_condition(
+    alert_rule_trigger: AlertRuleTrigger,
+) -> tuple[DataCondition, AlertRuleTriggerDataCondition] | None:
+    try:
+        alert_rule_detector = AlertRuleDetector.objects.get(
+            alert_rule=alert_rule_trigger.alert_rule
+        )
+    except AlertRuleDetector.DoesNotExist:
+        logger.exception(
+            "AlertRuleDetector does not exist",
+            extra={"alert_rule_id": alert_rule_trigger.alert_rule.id},
+        )
+        return None
+
+    threshold_to_condition = {
+        AlertRuleThresholdType.ABOVE.value: Condition.GREATER,
+        AlertRuleThresholdType.BELOW.value: Condition.LESS,
+        # TODO add ABOVE_AND_BELOW for anomaly detection
+    }
+
+    data_condition_group = alert_rule_detector.detector.workflow_condition_group
+    if not data_condition_group:
+        logger.warning(
+            "Could not find data_condition_group",
+            extra={"detector_id": alert_rule_detector.id},
+        )
+        return None
```

Comment:
should this just create the data condition group if it doesn't exist? (since this is a migration helper)